File: Astronomy-objc/Astronomy-objc/Camera.m
//
//  Camera.m
//  Astronomy-objc
//
//  Created by Dillon P on 4/26/20.
//  Copyright © 2020 Dillon's Lambda Team. All rights reserved.
//

#import "Camera.h"

@implementation Camera

- (instancetype)initWithIdentifier:(int)identifier
                              name:(NSString *)name
                           roverID:(int)roverID
                          fullName:(NSString *)fullName
{
    if (self = [super init]) {
        _identifier = identifier;
        _name = name;
        _roverID = roverID;
        _fullName = fullName;
    }
    return self;
}

@end

File: Astronomy-objc/Astronomy-objc/Model/Camera.swift
//
//  Camera.swift
//  Astronomy-objc
//
//  Created by Bradley Yin on 10/14/19.
//  Copyright © 2019 bradleyyin. All rights reserved.
//

import Foundation

//@objc(BYCamera)
class Camera: NSObject {
    let id: Int
    let name: String
    let roverId: Int
    let fullName: String
    
    @objc init(id: Int, name: String, roverId: Int, fullName: String) {
        self.id = id
        self.name = name
        self.fullName = fullName
        self.roverId = roverId
    }
}
